src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -281,6 +281,13 @@ def forward(
         cos, sin = self.rotary_emb(value_states, position_ids)
         query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin)
 
+        if q_len > 1:
+            # prefill
+            cache_position = torch.arange(cache_length, dtype=torch.int32, device=hidden_states.device)
+        else:
+            # decoding
+            cache_position = torch.tensor([cache_length - 1], dtype=torch.int32, device=hidden_states.device)
```

Comment:
I am very much supprised that this now works in torch compile with reduce overhead, as when testing this use to always create the same tensor (constant cache lenght) outputs were different. WOuld need investigation on which torch version supports this!